File: system/jv/abe/BELS_Base/BECS_Runtime.java
package abe.BELS_Base;

import java.util.*;

import abe.BEL_4_Base.BEC_6_6_SystemObject;
import abe.BEL_4_Base.BEC_5_4_LogicBool;
import abe.BEL_4_Base.BEC_6_11_SystemInitializer;

//This should be the var type to avoid unnecessary casting
//if we want to optimize "object" level calls we can put the signatures
//in this class and call directly...

//members array and mod for dynamic calls

public class BECS_Runtime {
    
    public static boolean isInitted = false;
   
    public static BEC_5_4_LogicBool boolTrue = new BEC_5_4_LogicBool(true);
    public static BEC_5_4_LogicBool boolFalse = new BEC_5_4_LogicBool(false);
    
    public static Map<String, Class> typeInstances;
    
    //for setting up initial instances
    public static BEC_6_11_SystemInitializer initializer;
    
    public static String platformName;
    
    public static String[] args;
    
    public static void init() { 
        if (isInitted) { return; }
        isInitted = true;
        BECS_Ids.callIds = new HashMap<String, Integer>();
        BECS_Ids.idCalls = new HashMap<Integer, String>();
        BECS_Ids.callIdCounter = 0;
        typeInstances = new HashMap<String, Class>();
        initializer = new BEC_6_11_SystemInitializer();
    }
    
}

File: system/cs/abe/BELS_Base/BECS_Runtime.cs
namespace abe.BELS_Base {

using System;
using System.Collections.Generic;
using abe.BEL_4_Base;

public class BECS_Runtime {
    
    public static bool isInitted = false;
    
    public static BEC_5_4_LogicBool boolTrue = new BEC_5_4_LogicBool(true);
    public static BEC_5_4_LogicBool boolFalse = new BEC_5_4_LogicBool(false);
    
    public static Dictionary<string, Type> typeInstances;
    
    //for setting up initial instances
    public static BEC_6_11_SystemInitializer initializer;
    
    public static string platformName;
    
    public static string[] args;
    
    public static void init() { 
        if (isInitted) { return; }
        isInitted = true;
        BECS_Ids.callIds = new Dictionary<string, int>();
        BECS_Ids.idCalls = new Dictionary<int, string>();
        BECS_Ids.callIdCounter = 0;
        typeInstances = new Dictionary<string, Type>();
        initializer = new BEC_6_11_SystemInitializer();
    }
    
}

}

